runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -1545,6 +1554,28 @@ TR::SymbolValidationManager::validateMethodFromSingleAbstractImplementerRecord(u
    return validateSymbol(methodID, definingClassID, method);
    }
 
+bool
+TR::SymbolValidationManager::validateDynamicMethodFromCallsiteIndex(uint16_t methodID, uint16_t beholderID, int32_t callsiteIndex, const char *signature, uint32_t length)
+   {
+   TR_OpaqueMethodBlock *method = getMethodFromID(methodID);
+   TR_OpaqueMethodBlock *dummyInvoke = _fej9->getMethodFromName("java/lang/invoke/MethodHandle", "linkToStatic", "([Ljava/lang/Object;)Ljava/lang/Object;");
+
+   TR_OpaqueClassBlock *beholder = getClassFromID(beholderID);
+   J9ROMClass *romClass = reinterpret_cast<J9Class *>(beholder)->romClass;
+
+   J9SRP                 *namesAndSigs = (J9SRP*)J9ROMCLASS_CALLSITEDATA(romClass);
+   J9ROMNameAndSignature *nameAndSig   = NNSRP_GET(namesAndSigs[callsiteIndex], J9ROMNameAndSignature*);
+   J9UTF8                *signatureUTF8 = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
+
+   int32_t signatureLength;
+   const char * linkToStaticSignature = _fej9->getSignatureForLinkToStaticForInvokeDynamic(_comp, signatureUTF8, signatureLength);
+
+   return (NULL != signature)
+          && (method == dummyInvoke)
+          && (signatureLength == length)
+          && (0 == strncmp(signature, linkToStaticSignature, length));
```

Comment:
I wasn't sure if the polymorphic signature of the dummy resolved `linkToStatic()` is something that is specified in the romclass of the beholder; if it is then you're right we don't need this. I had added this because if the polymorphic signature is something that is known only at runtime, then it would need to be explicitly stored as part of the validation.

Reply:
It's not specified directly, but it's derived from the signature used at the call site. In the case of unresolved `invokedynamic` we just add `Ljava/lang/Object;Ljava/lang/Object;` to the end of the parameters. Unresolved `invokehandle` is the same but it adds another extra `Ljava/lang/Object;` at the beginning of the parameters as well

Reply:
We definitely need the validations/relocations for resolved dispatch, but this PR is only for unresolved dispatch.

Reply:
>I.e., for unresolved dispatch, we don't generate TR_CallsiteTableEntryAddress and TR_MethodTypeTableEntryAddress relocations

Well.. this is true on x86; not sure about the other platforms though

Reply:
Yeah, and we don't even need the JITServer stream version bump if the as-yet unused relocations are left out

Reply:
I think the relocations will only work for resolved symrefs due to the `setCallsiteIndex()`/`setCpIndex()` step in `initializeCommonAOTRelocationHeader()`. They should really only be needed in cases where the symref was resolved anyway, since when it's unresolved, we generate code that resolves and finds the address at runtime, so what would we even relocate? (except maybe a helper call, but that's a different relo)